tests/error/errors_on_usage/for_new_var.err

```diff
@@ -1,7 +1,11 @@
-Guppy compilation failed. Error in file $FILE:8
+Error: Variable not defined (at $FILE:8:11)
+  | 
+6 |     for _ in xs:
+7 |         y = 5
+8 |     return y
+  |            ^ `y` might be undefined ...
+  | 
+6 |     for _ in xs:
+  |              -- ... if this expression is `False`
```

Comment:
this is a bit confusing, do we need to say "Falsey"...?

Reply:
I like the idea of differentiating between bools and values that are truthy. The problem is that we can no longer make this distinction here since we already inserted an implicit call to `__bool__`. 

An additional problem is that for loops are desugared to
```python
iter = xs.__iter__()
while True:
   b, iter = iter.__hasnext__()
   if b:
      x, it = iter.__next__()
   else:
      break
iter.__end__()
```
and the message actually points to the `while True` bit which actually *is* a bool.

Created issues #591 and #592 to figure out how to solve this